File: frontend/types/workflow.ts
/**
 * Type definitions for workflow data structures
 */

export interface WorkflowStep {
  id: string
  name: string
  description: string
  action_type: string
  tool: string
  parameters: Record<string, any>
  next_step_id?: string | null
}

export interface WorkflowBlueprintData {
  workflow_id: string
  goal: string
  steps: WorkflowStep[]
  edges: Array<{
    id: string
    source: string
    target: string
    type?: string
  }>
  nodes: Array<{
    id: string
    type: string
    position: { x: number; y: number }
    data: {
      label: string
      description: string
      tool: string
      action_type: string
    }
  }>
}

File: backend/schemas/workflow.py
"""
Pydantic schemas for workflow data structures
"""
from pydantic import BaseModel
from typing import List, Dict, Any, Optional


class WorkflowStep(BaseModel):
    """Individual step in a workflow"""
    id: str
    name: str
    description: str
    action_type: str  # e.g., "api_call", "data_transform", "notification"
    tool: str  # e.g., "airtable", "sendgrid", "webhook"
    parameters: Dict[str, Any]
    next_step_id: Optional[str] = None


class WorkflowBlueprint(BaseModel):
    """Complete workflow blueprint with React Flow compatible structure"""
    workflow_id: str
    goal: str
    steps: List[WorkflowStep]
    edges: List[Dict[str, Any]]  # React Flow edges format
    nodes: List[Dict[str, Any]]  # React Flow nodes format

